File: backend/src/main/java/info/gamewise/lor/videos/deckcodeextractor/DescriptionWithDeckCodeExtractor.java
package info.gamewise.lor.videos.deckcodeextractor;

import java.util.Optional;

public class DescriptionWithDeckCodeExtractor extends AbstractDeckCodeExtractor implements DeckCodeExtractor {

    private static final int MIN_CODE_LENGTH = 43;

    @Override
    public Optional<String> extract(String videoDescription) {
        return descriptionAsStream(videoDescription)
                .filter(this::filterDeckCode)
                .findFirst();
    }

    private boolean filterDeckCode(String word) {
        return word.length() >= MIN_CODE_LENGTH && !word.contains("http");
    }
}

File: backend/src/main/kotlin/info/gamewise/lor/videos/deckcodeextractor/DescriptionWithDeckCodeExtractor.kt
package info.gamewise.lor.videos.deckcodeextractor

import org.springframework.stereotype.Component

private const val MIN_CODE_LENGTH = 43

@Component
class DescriptionWithDeckCodeExtractor : AbstractDeckCodeExtractor(), DeckCodeExtractor {

    override fun extract(videoDescription: String): String? {
        return descriptionAsStream(videoDescription)
            .firstOrNull{ filterDeckCode(it) }
    }

    private fun filterDeckCode(word: String): Boolean {
        return word.length >= MIN_CODE_LENGTH && !word.contains("http")
    }

}